### ISSUE TITLE
Python's insecure_interop_test_InsecureInteropTest fails often.

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#8389]
https://grpc-testing.appspot.com/job/gRPC_master_linux/159/testReport/junit/(root)/python_linux_dbg/py27_test__insecure_interop_test_InsecureInteropTest/

seems to be the most problematic test of python

```
[95mTesting gRPC Python...
[0m[92mSUCCESS       _insecure_interop_test.InsecureInteropTest.testCancelAfterBegin
[0mRunning       _insecure_interop_test.InsecureInteropTest.testCancelAfterFirstResponse
[92mSUCCESS       _insecure_interop_test.InsecureInteropTest.testCancelAfterFirstResponse
[0mRunning       _insecure_interop_test.InsecureInteropTest.testClientStreaming
[92mSUCCESS       _insecure_interop_test.InsecureInteropTest.testClientStreaming
[0mRunning       _insecure_interop_test.InsecureInteropTest.testEmptyUnary
[92mSUCCESS       _insecure_interop_test.InsecureInteropTest.testEmptyUnary
[0mRunning       _insecure_interop_test.InsecureInteropTest.testLargeUnary
[92mSUCCESS       _insecure_interop_test.InsecureInteropTest.testLargeUnary
[0mRunning       _insecure_interop_test.InsecureInteropTest.testPingPong
[92mSUCCESS       _insecure_interop_test.InsecureInteropTest.testPingPong
[0mRunning       _insecure_interop_test.InsecureInteropTest.testServerStreaming
[92mSUCCESS       _insecure_interop_test.InsecureInteropTest.testServerStreaming
[0mRunning       _insecure_interop_test.InsecureInteropTest.testTimeoutOnSleepingServer
[91mERROR         _insecure_interop_test.InsecureInteropTest.testTimeoutOnSleepingServer
[0m8 tests finished:
    7 successful
    1 unsuccessful
    0 skipped
    0 expected failures
    0 unexpected successes
Interrupted Tests:
    []

Errors/Failures: 
[91m_insecure_interop_test.InsecureInteropTest.testTimeoutOnSleepingServer[0m
[1mtraceback:[0m
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/var/local/git/grpc/src/python/grpcio_tests/tests/interop/_interop_test_case.py", line 64, in testTimeoutOnSleepingServer
    methods.TestCase.TIMEOUT_ON_SLEEPING_SERVER.test_interoperability(self.stub, None)
  File "/var/local/git/grpc/src/python/grpcio_tests/tests/interop/methods.py", line 406, in test_interoperability
    _timeout_on_sleeping_server(stub)
  File "/var/local/git/grpc/src/python/grpcio_tests/tests/interop/methods.py", line 274, in _timeout_on_sleeping_server
    next(response_iterator)
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/grpc/_channel.py", line 344, in next
    return self._next()
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/grpc/_channel.py", line 324, in _next
    raise self
_Rendezvous: <_Rendezvous of RPC that terminated with (StatusCode.UNKNOWN, Cancelled before creating subchannel)>
```

**Comment by dgquintas** [comment]
Seen again: https://grpc-testing.appspot.com/job/gRPC_master_linux/730/testReport/(root)/python_linux_dbg/py27_test_interop__insecure_interop_test_InsecureInteropTest/

**Comment by nathanielmanistaatgoogle** [comment]
@dgquintas: this seems for now to be a gRPC Core issue - I am able to reproduce the problem (nondeterministically with occurrence around one in every 💯 test runs) with Python code as simple as
```py
next(stub.FullDuplexCall(request_iterator, timeout=0.001))
```
(in which `request_iterator` is an object that looks like an iterator but blocks calls to its `next` method for the duration of the defect reproduction). 

Are there tests for gRPC Core that verify that such an extremely short RPC does not terminate with code `UNKNOWN` and details `Cancelled before creating subchannel`?

**Comment by dgquintas** [comment]
Well, we have one with an infinitely negative deadline :) https://github.com/grpc/grpc/blob/master/test/core/end2end/tests/negative_deadline.c that asserts the result is a DEADLINE_EXCEEDED. I'm investigating.

**Comment by nathanielmanistaatgoogle** [comment]
I can confirm that [pull request 9022](https://github.com/grpc/grpc/issues/9022) solves some but not all of `_insecure_interop_test`'s flakiness and that I am investigating a separate defect with a separate root cause.

**Comment by dgquintas** [comment]
@nathanielmanistaatgoogle the fix was re-introduced in #9089 . Can you confirm the issue has been fixed?

**Comment by nathanielmanistaatgoogle** [comment]
@dgquintas: yes, I consider that particular failure resolved.

I think that this issue will be fixed following integration into master of 8b223e2986fe92e93aa93b8524d25fbc6d3989b2.

**Comment by nathanielmanistaatgoogle** [comment]
Closing now that 8b223e2986fe92e93aa93b8524d25fbc6d3989b2 is on `master`.
